README.md

```diff
@@ -32,6 +32,8 @@ General guidelines for Hellbender developers
 
 * Untested code is considered **non-existent** and thus is subject to removal at any time (exception is main methods, or super corner conditions, or `toString()` code).
 
+* Each tool should have at least one good end-to-end integration test with a check for expected output, plus high-quality unit tests for all non-trivial utility methods/classes used by the tool. Although we have no specific coverage target, coverage should be extensive enough that if tests pass, the tool is guaranteed to be in a usable state.
```

Comment:
@nh13 Yes, we need your help. I went through all classes with 0% coverage. Some of them can indeed be tested by simple integration tests. Testing most of those classes however requires a good understanding of what the algorithm does and multiple tests need to be written to account for all those cases. 

Below, all classes marked as 'non clp' must be unit tested. Only those marked as 'simple tests OK' are really possible to meaningfully test by a single input/output check. 

```
FixMateInformation	clp	
ReorderSam	clp	
FilterReads	clp	
RevertOriginalBaseQualitiesAndAddMateCigar	clp	
AddOrReplaceReadGroups	clp	
BuildBamIndex	clp	simple tests OK
DownsampleSam	clp	
ReplaceSamHeader	clp	simple tests OK
BamToBfq	clp	TO DELETE ?
SortSam	clp	simple tests OK
FilterReads.Filter	clp	
CalculateReadGroupChecksum	clp	simple tests OK
CompareSAMs	clp	simple tests OK
CheckTerminatorBlock	clp	
BamIndexStats	clp	simple tests OK
CollectWgsMetrics	clp	simple tests OK
CollectJumpingLibraryMetrics	clp	
TargetMetricsCollector	not clp	
HsMetricCollector	not clp	
CollectTargetedMetrics	not clp	
TargetedPcrMetricsCollector	not clp	
CalculateHsMetrics	clp	simple tests OK
CountingFilter	not clp	
CollectTargetedPcrMetrics	clp	simple tests OK
CountingMapQFilter	not clp	
CountingPairedFilter	not clp	
CountingDuplicateFilter	not clp	
TargetedPcrMetrics	class without code	nothing to test
HsMetrics	class without code	nothing to test
TargetMetrics	class without code	nothing to test
JumpingLibraryMetrics	class without code	nothing to test
CollectGcBiasMetrics	clp	
CollectOxoGMetrics	clp	
QualityScoreDistribution	clp	simple tests OK
CollectQualityYieldMetrics	clp	simple tests OK
MeanQualityByCycle	clp	simple tests OK
CollectBaseDistributionByCycle	clp	simple tests OK
CollectMultipleMetrics	clp	simple tests OK
CompareMetrics	clp	simple tests OK
GcBiasDetailMetrics	class without code	nothing to test
GcBiasSummaryMetrics	class without code	nothing to test
BaseDistributionByCycleMetrics	class without code	nothing to test
ClusterIntensityFileReader	not clp	
MultiTileParser	not clp	
FourChannelIntensityData	not clp	
IlluminaTextIterator	not clp	
MultiTileBclParser	not clp	
Casava18ReadNameEncoder	not clp	
IntensityChannel	class without code	nothing to test
MultiTileLocsParser	not clp	
MultiTileFilterParser	not clp	
IlluminaBasecallsToFastq	clp	
MarkIlluminaAdapters	clp	
GatherVcfs	clp	needs to test all cases 
LiftoverVcf	clp	simple tests OK - needs to test multiple cases
MergeVcfs	clp	
MakeSitesOnlyVcf	clp	simple tests OK but needs multiple cases
RenameSampleInVcf	clp	simple tests OK 
VcfToIntervalList	clp	simple tests OK
IntervalListTools	clp	
LiftOverIntervalList	clp	simple tests OK
EstimateLibraryComplexity	clp	
CollectHiSeqXPfFailMetrics	clp	
BclIndexReader	not clp	
GenotypeConcordanceStates	not clp	
ConvertSequencingArtifactToOxoG	clp	not sure
SequencingArtifactMetrics	not clp	
MemoryBasedReadEndsForMarkDuplicatesMap	not clp	
NormalizeFasta	clp	simple tests OK
ExtractSequences	clp	simple tests OK
```